tor/controller_test.go

```diff
@@ -359,3 +359,33 @@ func TestReconnectSucceed(t *testing.T) {
 	// Check that the connection has been updated.
 	require.NotEqual(t, proxy.clientConn, c.conn)
 }
+
+// TestParseTorReply tests that Tor replies are parsed correctly.
+func TestParseTorReply(t *testing.T) {
+	testCase := []struct {
+		reply          string
+		expectedParams map[string]string
+	}{
+		{
+			// Test a regular reply.
+			reply:          `VERSION Tor="0.4.7.8"`,
+			expectedParams: map[string]string{"Tor": "0.4.7.8"},
+		},
+		{
+			// Test a reply with multiple values, one of them containing spaces.
+			reply:          `AUTH METHODS=COOKIE,SAFECOOKIE,HASHEDPASSWORD COOKIEFILE="/path/with/spaces/Tor Browser/control_auth_cookie"`,
+			expectedParams: map[string]string{"METHODS": "COOKIE,SAFECOOKIE,HASHEDPASSWORD", "COOKIEFILE": "/path/with/spaces/Tor Browser/control_auth_cookie"},
+		},
+		{
+			// Test a multiline reply.
+			reply: `ServiceID=id
+OK`,
```

Comment:
This looks like a formatting error. Should we use a non-literal string and escape the newline with `\n` or `\r\n` instead?